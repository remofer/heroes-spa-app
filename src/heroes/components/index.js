export * from "./HeroItem";
export * from "./HeroList";
export * from "./HeroCharacters";
export * from "./Hero";