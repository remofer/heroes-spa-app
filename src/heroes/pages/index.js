export * from "./MarvelPages";
export * from "./DCPages";
export * from "./ErrorPage";
export * from "./HeroPage";
export * from "./SearchPage";
