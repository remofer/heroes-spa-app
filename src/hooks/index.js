export * from "./useForm";
export * from "./useCustomNavigation";
