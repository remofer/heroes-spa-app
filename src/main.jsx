import React from "react";
import { createRoot } from "react-dom/client";
import { HeroesApp } from "./HeroesApp";

import "./styles.css";

const root = createRoot(document.getElementById("root"));
root.render(
  <React.StrictMode>
    <HeroesApp />
  </React.StrictMode>
);
