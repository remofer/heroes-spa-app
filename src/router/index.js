export * from "./HeroRouter";
export * from "./PrivateRoute";
export * from "./PublicRoute";
export * from "./HeroRouter";
export * from "./AppRouter";
