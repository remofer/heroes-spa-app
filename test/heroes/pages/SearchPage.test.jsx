import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../src/heroes";

const mockUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUseNavigate,
}));

describe("Pruebas en la p√°gina de <SearchPage/>", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("Debe de mostrarse correctamente con los valores por defecto ", () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );
    // screen.debug();
    expect(container).toMatchSnapshot();
  });

  test("Debe de a Batman y el input con el valor del queryString", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <SearchPage />
      </MemoryRouter>
    );
    const inputValue = screen.getByRole("textbox");
    expect(inputValue.value).toBe("batman");
    const img = screen.getByRole("img");
    expect(img.src).toContain("/assets/heroes/dc-batman.jpg");
    const alert = screen.getByLabelText("alert-danger");
    expect(alert.style.display).toBe("none");
    // screen.debug();
  });

  test("Debe mostrar un error si no se encuentre el hero (batman123)", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman123"]}>
        <SearchPage />
      </MemoryRouter>
    );
    const alert = screen.getByLabelText("alert-danger");
    expect(alert.style.display).not.toBe("none");
  });

  test("Debe llamar el navigate a la pantalla nueva", () => {
    render(
      <MemoryRouter initialEntries={["/search"]}>
        <SearchPage />
      </MemoryRouter>
    );
    const inputValue = screen.getByRole("textbox");
    fireEvent.change(inputValue, {
      target: { name: "searchTxt", value: "batman" },
    });
    const form = screen.getByRole("form");
    fireEvent.submit(form);

    expect(mockUseNavigate).toHaveBeenCalledWith("?q=batman");
  });
});
