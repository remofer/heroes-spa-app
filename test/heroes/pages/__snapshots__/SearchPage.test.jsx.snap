// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la p√°gina de <SearchPage/> Debe de mostrarse correctamente con los valores por defecto  1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchTxt"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeInRight"
      >
        Search Hero
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeInRight"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
